lib/oofatfs/ff.c

```diff
@@ -3382,7 +3382,11 @@ FRESULT f_read (
             if (!sect) ABORT(fs, FR_INT_ERR);
             sect += csect;
             cc = btr / SS(fs);                  /* When remaining bytes >= sector size, */
-            if (cc) {                           /* Read maximum contiguous sectors directly */
+            if (cc
+#if _FS_DISK_READ_ALIGNED
```

Comment:
@tannewt you may want to scrutinize this part.  It's fixing #2332

Reply:
As we chatted about. We'll leave #2332 open with ideas for a proper fix.